Content.Server/GameObjects/Components/Power/PowerCellSlotComponent.cs

```diff
@@ -0,0 +1,244 @@
+            {
+                if (!user.TryGetComponent(out HandsComponent? hands) || !hands.PutInHand(cell.Owner.GetComponent<ItemComponent>()))
+                {
+                    cell.Owner.Transform.Coordinates = user.Transform.Coordinates;
+                }
+            }
+            else
+            {
+                cell.Owner.Transform.Coordinates = Owner.Transform.Coordinates;
+            }
+
+            if (playSound && CellRemoveSound != null)
+            {
+                EntitySystem.Get<AudioSystem>().PlayFromEntity(CellRemoveSound, Owner);
+            }
+            Owner.SendMessage(this, new PowerCellChangedMessage(true));
+            return cell;
+        }
+
+        /// <summary>
+        /// Tries to insert the given cell into this component. The cell will be put into the container of this component.
+        /// </summary>
+        /// <param name="cell">The cell to insert.</param>
+        /// <param name="playSound">Should <see cref="CellInsertSound"/> be played upon insertion?</param>
+        /// <returns>True if insertion succeeded; false otherwise.</returns>
+        public bool InsertCell(IEntity cell, bool playSound = true)
+        {
+            if (Cell != null) return false;
+            if (!cell.TryGetComponent<PowerCellComponent>(out var cellComponent)) return false;
+            if (cellComponent.CellSize != SlotSize) return false;
+            if (!_cellContainer.Insert(cell)) return false;
+            //Dirty();
+            if (playSound && CellInsertSound != null)
+            {
+                EntitySystem.Get<AudioSystem>().PlayFromEntity(CellInsertSound, Owner);
+            }
+            Owner.SendMessage(this, new PowerCellChangedMessage(false));
+            return true;
+        }
+
+        [Verb]
+        public sealed class EjectCellVerb : Verb<PowerCellSlotComponent>
+        {
+            protected override void GetData(IEntity user, PowerCellSlotComponent component, VerbData data)
+            {
+                if (!component.ShowVerb || !ActionBlockerSystem.CanInteract(user))
+                {
+                    data.Visibility = VerbVisibility.Invisible;
+                    return;
+                }
+
+                if (component.Cell == null)
+                {
+                    data.Text = Loc.GetString("Eject cell (cell missing)");
+                }
+                else
+                {
+                    data.Text = Loc.GetString("Eject cell");
+                }
+
+                if (component.Cell == null || !component.CanRemoveCell)
+                {
+                    data.Visibility = VerbVisibility.Disabled;
+                }
+            }
+
+            protected override void Activate(IEntity user, PowerCellSlotComponent component)
+            {
+                component.EjectCell(user);
+            }
+        }
+
+        void IMapInit.MapInit()
+        {
+            if (_startEmpty || _cellContainer.ContainedEntity != null)
```

Comment:
Having a test for this would be nice but not the end of the world to leave it out.